### PR TITLE
Search component

### PR DESCRIPTION
**Pull request by DwinaTech**

### CODE REVIEW

File: frontend/src/Components/Organisation/Option.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { MenuItem } from 'material-ui/Menu';
+
+// ========= Selected List ===========//
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/Option.js]: What's the reason for this comment? if you realise you need to make the name of the component better, just change the name from "Option" to "SelectList"

File: frontend/src/Components/Organisation/Search.js [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+import 'react-select/dist/react-select.css';
+import Option from './Option';
+import searchStyle from './searchStyle';
+
+const organisations = [
+  { postCode: 'H2 1TH', address: 'Bermondsey', day: 'Monday' },
+  { postCode: 'H2 2TH', address: '2 Bermondsey', day: 'Tuesday' },
+  { postCode: 'H2 3TH', address: '3 Bermondsey', day: 'Wednesday' },
+];
+const days = [
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/Search.js]: what is this? name the array more descriptive, it's clearly not a list of days

File: frontend/src/Components/Organisation/Search.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+import ArrowDropUpIcon from 'material-ui-icons/ArrowDropUp';
+import ClearIcon from 'material-ui-icons/Clear';
+import Chip from 'material-ui/Chip';
+import Select from 'react-select';
+import 'react-select/dist/react-select.css';
+import Option from './Option';
+import searchStyle from './searchStyle';
+
+const organisations = [
+  { postCode: 'H2 1TH', address: 'Bermondsey', day: 'Monday' },
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/Search.js]: What is this exactly? why do these organisations only have 3 properties?

File: frontend/src/Components/Organisation/Search.js [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+const days = [
+  { postCode: 'H2 1TH', address: 'Bermondsey', day: 'Monday' },
+  { postCode: 'H2 2TH', address: '2 Bermondsey', day: 'Tuesday' },
+  { postCode: 'H2 3TH', address: '3 Bermondsey', day: 'Wednesday' },
+].map(organisation => ({ // === organisation by day =========
+  value: organisation.day,
+  label: organisation.day,
+}));
+
+// ========== Auto Compelet Search Field
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/Search.js]: This file seems to be too long and have too many things, break it in to at least one `helpers.js` or `utilities.js` and the separate components per file, import required functions and use them when needed

File: frontend/src/Components/Organisation/Search.js [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+            <strong key={String(index)} style={{ fontWeight: 500 }}>
+              {part.text}
+            </strong>
+          )
+        )}
+      </div>
+    </MenuItem>
+  );
+}
+// ========== Auto Compelet Search Warpper
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/Search.js]: ditto - why comment when you can just choose a better name? comment should clarify and all comments should have a consistent style.. this particular one is not a convention I am familiar with

File: frontend/src/Components/Organisation/Search.js [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+        count < 5 && organisation.postCode.toLowerCase().slice(0, inputLength) === inputValue;
+
+      if (keep) {
+        count += 1;
+      }
+
+      return keep;
+    });
+}
+// ###################################
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/Search.js]: is this a particular style of comment? :)

File: frontend/src/Components/Organisation/Search.js [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+
+        return <div className="Select-value">{children}</div>;
+      }}
+      {...other}
+    />
+  );
+}
+// ####################################
+// ========== Search Component ========
+class Search extends React.Component {
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/Search.js]: If a component file is getting to 150 lines... It's too long. Separate in to a different file, easier to read/maintain

File: frontend/src/Components/Organisation/Search.js [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+    });
+  };
+
+  handleChangeSingle = single => {
+    this.setState({
+      single,
+    });
+  };
+
+  handleChange = e => {
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/Search.js]: handleChange is not descriptive. Can you tell by reading the function name? or do you have to read all the file to understand what this is handling?

File: frontend/src/Components/Organisation/organisation.css [hunk 9]
```diff
@@ -365,4 +365,41 @@
 }
 .org-content .not-include-in-add-org{
   display: none !important;
+}
+/*===================================================*/
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/organisation.css]: CSS files can be separated to different files too, no need for separating comments like this. Just make a different CSS file, and import the relevant part to the relevant component

File: frontend/package.json [hunk 10]
```diff
@@ -3,11 +3,15 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "autosuggest-highlight": "^3.1.1",
```
Comment (kkarimi) [inline frontend/package.json]: If requirements change, yarn lock needs to be changed too. Keep an eye on your PR and you will see that it the lint/test stage complains that it is broken for this reason

File: frontend/src/Components/Organisation/Search.js (hunk 3 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: just to test, I can search on list of organisations by post code, then after select the post code should select the full address of the organisation

File: frontend/src/Components/Organisation/Search.js (hunk 4 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: alright

Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: Done

File: frontend/src/Components/Organisation/Search.js (hunk 5 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: Done

File: frontend/src/Components/Organisation/Search.js (hunk 6 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: Removed

File: frontend/src/Components/Organisation/Search.js (hunk 7 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: Removed

File: frontend/src/Components/Organisation/Option.js (hunk 1 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Option.js]: this component was in search component I forget to remove it after I made the component individual one. it removed now.

File: frontend/src/Components/Organisation/Search.js (hunk 2 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: :smile: corrected

File: frontend/src/Components/Organisation/Search.js (hunk 8 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: changed to logic name

File: frontend/src/Components/Organisation/organisation.css (hunk 9 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/organisation.css]: Done

File: frontend/package.json (hunk 10 above)
Comment (DwinaTech) [reply frontend/package.json]: alright

File: frontend/src/Components/Organisation/Search.js (hunk 2 above)
Comment (kkarimi) [reply frontend/src/Components/Organisation/Search.js]: it's still not just days.. there's a map part too.

File: frontend/package.json (hunk 10 above)
Comment (kkarimi) [reply frontend/package.json]: still not fixed, do you know how to check circle CI? for example [here](https://circleci.com/gh/CodeYourFuture/directory-of-services/209) is the build fail

Comment (DwinaTech) [reply frontend/package.json]: I am straggling with this point actually

Comment (DwinaTech) [reply frontend/package.json]: fixed

File: frontend/src/Components/Organisation/Search.js (hunk 2 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: corrected

Comment (kkarimi) [reply frontend/src/Components/Organisation/Search.js]: @DwinaTech So...it's still called  `days` but this variable..... but are these maybe `weekDays`? `onlyMonToWedsDays`? let me know when you know!

Comment (DwinaTech) [reply frontend/src/Components/Organisation/Search.js]: Added

**Pull request merged**